dns-integrator-operator/src/charm.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""DNS Integrator Charm."""
+
+import logging
+import typing
+import hashlib
+import uuid
+
+import ops
+from charms.bind.v0 import dns_record
+
+logger = logging.getLogger(__name__)
+
+VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
+
+
+class DnsIntegratorOperatorCharm(ops.CharmBase):
+    """Charm the service."""
+
+    def __init__(self, *args: typing.Any):
+        """Construct.
+
+        Args:
+            args: Arguments passed to the CharmBase parent constructor.
+        """
+        super().__init__(*args)
+        self.dns_record = dns_record.DNSRecordRequires(self)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+
+    def _on_config_changed(self, _: ops.ConfigChangedEvent) -> None:
+        """Handle changes in configuration."""
+        self.unit.status = ops.MaintenanceStatus("Configuring charm")
+        logger.debug("UPDATE INTEGRATOR CONFIG")
+        self._update_relations()
+        self.unit.status = ops.ActiveStatus()
+
+    def _update_relations(self) -> None:
+        """Update all DNS data for the existing relations."""
+        if not self.model.unit.is_leader():
+            return
+        try:
+            for relation in self.model.relations[self.dns_record.relation_name]:
+                self.dns_record.update_relation_data(
+                    relation, self._get_dns_record_requirer_data()
+                )
+        except ops.model.ModelError as e:
+            logger.error("ERROR while updating relation data: %s", e)
```

Comment:
In this case, the charm will be still active. Is this ok?

Reply:
I feel like it's okay. I want to the charm to be waiting to retry or for new data ?

Reply:
I don't want to go to an error state and I'm not sure if blocked is correct here since it may come from an issue non operator related

Reply:
I see. What do you think about doing something similar to the lego provider? They change the status message like this:
https://github.com/canonical/lego-operator/blob/6a91a06cc223bd71cebe029b68268f6967a90bb1/src/charm.py#L171